packages/nocodb/src/db/BaseModelSqlv2.ts

```diff
@@ -4512,6 +4514,20 @@ class BaseModelSqlv2 {
     return true;
   }
 
+  /*
+   *  Utility method to validate database constraints
+   */
+  protected validateConstraints(
+    column: Column<any>,
+    data: Record<string, any>,
+  ) {
+    if (column.dtxp && column.dtxp < data[column.title]?.length) {
```

Comment:
Scale can be there for integer type as well. And if the value is a number/boolean `length` property won't be there.

Reply:
If this pattern looks good, I can add in checks for all the field types

Reply:
in the case of number, we need to confirm what it means(scale value), like the number of digits or bits

Reply:
You can fix it for string at the moment by checking `typeof data[column.title] === 'string'`